lib/matplotlib/quiver.py

```diff
@@ -487,6 +502,25 @@ def __init__(self, ax, *args,
         self.angles = angles
         self.width = width
 
+        # Checks the boundaries of head_pos if outside range default of 0.5
+        if (head_pos == 0.0) or (head_pos == "tail"):
+            self.head_pos = 0.0
+        elif (head_pos == 1.0) or (head_pos == "tip"):
+            self.head_pos = 1.0
+        elif (head_pos == "mid") or (head_pos == "middle"):
+            self.head_pos = 0.5
+        elif type(head_pos) == float:
```

Comment:
If you combine this into:
```python
elif type(head_pos) == float and 0.0 < head_pos < 1.0:
```
then you only need the single final `else` to raise `ValueError`.